src/en/clowncorps/src/eu/kanade/tachiyomi/extension/en/clowncorps/ClownCorps.kt

```diff
@@ -0,0 +1,241 @@
+    }
+
+    private fun parseDate(dateStr: String): Long {
+        return try {
+            dateFormat.parse(dateStr)!!.time
+        } catch (_: ParseException) {
+            0L
+        }
+    }
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("MMMM dd, yyyy hh:mm aa", Locale.ENGLISH)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val doc = response.asJsoup()
+        val pages = mutableListOf<Page>()
+
+        val image = doc.selectFirst("#comic img") ?: return pages
+
+        val url = image.attr("src")
+        pages.add(Page(0, "", url))
+
+        if (showAuthorsNotesPref()) {
+            val title = image.attr("title")
+
+            // Ignore chapters that don't really have author's notes
+            val ignoreRegex = Regex("""^chapter \d+ page \d+$""", RegexOption.IGNORE_CASE)
+            if (ignoreRegex.matches(title)) return pages
+
+            val localURL = TextInterceptorHelper.createUrl("Author's Notes from $CREATOR", title)
+            val textPage = Page(pages.size, "", localURL)
+            pages.add(textPage)
+        }
+
+        return pages
+    }
+
+    override fun imageUrlParse(response: Response) =
+        throw UnsupportedOperationException()
+
+    override fun latestUpdatesParse(response: Response) =
+        throw UnsupportedOperationException()
+
+    override fun latestUpdatesRequest(page: Int) =
+        throw UnsupportedOperationException()
+
+    override fun mangaDetailsParse(response: Response) =
+        throw UnsupportedOperationException()
+
+    override fun popularMangaParse(response: Response) =
+        throw UnsupportedOperationException()
+
+    override fun popularMangaRequest(page: Int) =
+        throw UnsupportedOperationException()
+
+    override fun searchMangaParse(response: Response) =
+        throw UnsupportedOperationException()
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) =
+        throw UnsupportedOperationException()
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private fun showAuthorsNotesPref() = preferences.getBoolean(SETTING_KEY_SHOW_AUTHORS_NOTES, false)
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        val authorsNotesPref = SwitchPreferenceCompat(screen.context).apply {
+            key = SETTING_KEY_SHOW_AUTHORS_NOTES
+            title = "Show author's notes"
+            summary =
+                "Enable to see the author's notes at the end of chapters (if they're there)."
+            setDefaultValue(false)
+        }
+        screen.addPreference(authorsNotesPref)
+
+        // I couldn't find a way to create a simple button, so here's a workaround that uses
+        // a MultiSelectListPreference with a single option as a kind of confirmation window.
+        val clearCachePref = MultiSelectListPreference(screen.context).apply {
```

Comment:
Using `SwitchPreferenceCompat` is enough (just don't confirm). And you don't need to specify a key for `SwitchPreferenceCompat` if you use it like a button.

Reply:
It works just as well, yes, but I feel like it's a less clear user experience.
This UX signifies an option that toggles something, while using a pop-up, like I did previously, makes it feel completely seamless to the user.

I have tested this with some actual users, who all confirmed this.
I posted GIFs of both the toggle button (1), and the pop-up (2), which the question "What do you find clearer?"

> the second one because a toggle for something that isnt a toggle is weird

> By far the second option
> There is no world in which the first makes any sense
> I would consider option 1 broken software lol

Reply:
Yeah but I'd prefer keeping the code as simple as possible. People don't visit the settings page very often, so it shouldn't need such a complicated workaround.

Reply:
I would say having a good ux is important too

Reply:
I feel like we shouldn't compromise on UX, especially when the alternative option is as confusing as this.
And honestly, this code is not bad at all.
The only "odd" part is this:
```kotlin
setOnPreferenceChangeListener { _, newValue ->
    val checkValue = newValue as Set<*>
    if (checkValue.contains(VALUE_CONFIRM)) {
```
And this was adapted from how the MangaDex source does it: https://github.com/keiyoushi/extensions-source/blob/dfb59ebae0c8fa2b1a118e86b5bc5be8a183f3e6/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDex.kt#L684-L685

Reply:
Sure.